relayer/application_relayer.go

```diff
@@ -102,30 +106,77 @@ func newApplicationRelayer(
 	checkpointManager := checkpoint.NewCheckpointManager(logger, db, sub, relayerID, startingHeight)
 	checkpointManager.Run()
 
-	ar := applicationRelayer{
+	ar := ApplicationRelayer{
 		logger:            logger,
 		metrics:           metrics,
 		network:           network,
 		messageCreator:    messageCreator,
-		responseChan:      responseChan,
 		sourceBlockchain:  sourceBlockchain,
 		relayerID:         relayerID,
 		signingSubnetID:   signingSubnet,
 		warpQuorum:        quorum,
 		checkpointManager: checkpointManager,
+		currentRequestID:  rand.Uint32(), // TODONOW: pass via ctor
+		lock:              &sync.RWMutex{},
 	}
 
 	return &ar, nil
 }
 
-func (r *applicationRelayer) relayMessage(
-	unsignedMessage *avalancheWarp.UnsignedMessage,
+// Process [msgs] at height [height] by relaying each message to the destination chain.
+// Checkpoints the height with the checkpoint manager when all messages are relayed.
+func (r *ApplicationRelayer) ProcessHeight(height uint64, msgs []messages.MessageHandler) error {
+	var eg errgroup.Group
+	for _, msg := range msgs {
+		eg.Go(func() error {
+			return r.ProcessMessage(msg)
```

Comment:
Upgraded to go 1.22 so that this closure captures the value of `msg` on each iteration, rather than the variable itself.

Reply:
That's an interesting change for a language upgrade. I'm surprised if this change doesn't break some apps.

Reply:
@cam-schultz There's a flag you can set in `1.21` that enables this feature, so we can keep our versions consistent.

Reply:
https://tip.golang.org/wiki/LoopvarExperiment

Reply:
Thanks for pointing that out @geoff-vball . I elected to go with the workaround of assigning a local variable in each iteration rather than include that experimental flag in the build, since we're trying to lock into stable language features.